### PR TITLE
Refactor torch.onnx documentation

### LINKED ISSUE #108274: API: Restructure ONNX API doc with clear separation between TorchScript-based and FX-based explorer

### PR DESCRIPTION
**Pull request by thiagocrepaldi**
* Distinguish both TorchScript-based exporter (`torch.onnx.export`) and the TorchDynamo-based exporter (`torch.onnx.dynamo_export`) exporters
* Merge ONNX diagnostics page with the exporter page
* Add initial version of a quick overview on the new exporter
* Updates `torch.compiler.html` with the right page for the ONNX Runtime backend for `torch.compile`
* Renamed doc files to clearly identify files belonging to the legacy and newer onnx exporters

Fixes #108274

https://docs-preview.pytorch.org/pytorch/pytorch/108379/index.html

### CODE REVIEW

**Comment by thiagocrepaldi** [comment]
Suppressing a weird bc_linter error after renaming 2 internal infra scripts. The original error are shown below

```bash
fatal: path 'docs/source/scripts/onnx/build_onnx_diagnostics_rules_md.py' does not exist in '4fa99eba80d2dc64b0b4e1366214baccf3a8042f'
fatal: path 'docs/source/scripts/onnx/build_onnx_dynamo_diagnostics_rules_md.py' exists on disk, but not in '0f1a225f33b6e3966eda02baec5154fdc39a18dd'
fatal: path 'docs/source/scripts/onnx/build_onnx_supported_aten_op_csv_table.py' does not exist in '4fa99eba80d2dc64b0b4e1366214baccf3a8042f'
fatal: path 'docs/source/scripts/onnx/build_onnx_torchscript_supported_aten_op_csv_table.py' exists on disk, but not in '0f1a225f33b6e3966eda02baec5154fdc39a18dd'
Warning: Function gen_docs: function deleted
Warning: Function main: function deleted
Warning: Function main: function deleted
```

**Comment by justinchuby** [comment]
Prob need to mention dependencies required (onnxscript) and how to install them?

File: docs/source/onnx_dynamo.rst [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+As mentioned before, the ONNX exporter leverages TorchDynamo engine to hook into Python's frame evaluation API
+and dynamically rewrite its bytecode into an FX Graph.
+The resulting FX Graph is then polished before it is finally translated into an ONNX graph.
+
+The main advantage of this approach is that the `FX graph <https://pytorch.org/docs/stable/fx.html>_` is captured using
+bytecode analysis that preserves the dynamic nature of the model instead of using traditional static tracing techniques.
+
+The exporter is designed to be modular and extensible. It is composed of the following components:
+
+  - **ONNX Registry**: A registry of ONNX operators and functions.
```
Comment (wschin) [inline docs/source/onnx_dynamo.rst:22]: Add link to those classes?

File: docs/source/onnx_dynamo.rst [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+A simple example
+----------------
+
+See below a demonstration of exporter API in action with a simple image classifier model as example:
+
+.. code-block:: python
+
+  import torch
+
+  class MNISTModel(torch.nn.Module):
```
Comment (wschin) [inline docs/source/onnx_dynamo.rst:52]: I am wondering if we should use MLP as the first example.

File: docs/source/onnx_dynamo.rst [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+        output = torch.nn.functional.log_softmax(tensor_x, dim=1)
+        return output
+
+  tensor_x = torch.rand((64, 1, 28, 28), dtype=torch.float32)
+  export_output = torch.onnx.dynamo_export(MNISTModel(), tensor_x)
+
+As the code above shows, all you need is to provide ``torch.onnx.dynamo_export`` with an instance of the model and its input.
+The exporter will then return an instance of ``torch.onnx.ExportOutput`` that contains the exported ONNX graph along with extra information.
+
+The in-memory model is available at ``export_output.model_proto`` following `ONNX IR spec <https://github.com/onnx/onnx/blob/main/docs/IR.md>`_,
```
Comment (wschin) [inline docs/source/onnx_dynamo.rst:79]:
```suggestion
The in-memory model (type: onnx.ModelProto) is available at ``export_output.model_proto`` following `ONNX IR spec <https://github.com/onnx/onnx/blob/main/docs/IR.md>`_,
```

File: docs/source/onnx_dynamo.rst (hunk 2 above)
Comment (thiagocrepaldi) [reply docs/source/onnx_dynamo.rst:52]: What would change? The ultimate goal for the first example is show how to use the public API. The actual model architecture is kind of secondary given the promise of "work (in almost) every model"

File: docs/source/onnx_dynamo.rst (hunk 1 above)
Comment (thiagocrepaldi) [reply docs/source/onnx_dynamo.rst]: good idea.

**Comment by justinchuby** [comment]
1. Is this formatting intended (`**`)?
    image

2. Is the blue box intended?
    image

3. Link rendering: FX graph
    image

File: docs/source/onnx_dynamo.rst [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+.. contents:: :local:
+    :depth: 3
+
+.. warning::
+  The ONNX exporter for TorchDynamo is under active development and is subject to rapid change.
+
+Overview
+--------
+
+As mentioned before, the ONNX exporter leverages TorchDynamo engine to hook into Python's frame evaluation API
```
Comment (justinchuby) [inline docs/source/onnx_dynamo.rst:16]:
```suggestion
The ONNX exporter leverages TorchDynamo engine to hook into Python's frame evaluation API
```

nit: conciseness

File: docs/source/onnx_dynamo.rst [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+The main advantage of this approach is that the `FX graph <https://pytorch.org/docs/stable/fx.html>_` is captured using
+bytecode analysis that preserves the dynamic nature of the model instead of using traditional static tracing techniques.
+
+The exporter is designed to be modular and extensible. It is composed of the following components:
+
+  - **ONNX Exporter**: :class:`Exporter` main class that orchestrates the export process.
+  - **ONNX Export Options**: :class:`ExportOptions` has a set of options that control the export process.
+  - **ONNX Registry**: :class:`OnnxRegistry` is the registry of ONNX operators and functions.
+  - **FX Graph Extractor**: :class:`FXGraphExtractor` extracts the FX graph from the PyTorch model.
+  - **Fake Mode**: :class:`ONNXFakeContext` is a context manager that enables fake mode for large scale models.
```
Comment (justinchuby) [inline docs/source/onnx_dynamo.rst:29]: Links for these two are not working: 
image

File: docs/source/onnx_dynamo.rst [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+  - **ONNX Export Output**: :class:`ExportOutput` is the output of the exporter that contains the exported ONNX graph and diagnostics.
+  - **ONNX Export Output Serializer**: :class:`ExportOutputSerializer` serializes the exported model to a file.
+  - **ONNX Diagnostic Options**: :class:`DiagnosticOptions` has a set of options that control the diagnostics emitted by the exporter.
+
+Dependencies
+------------
+
+The ONNX exporter depends on extra Python packages not available through PyTorchs's `requirements.txt` file:
```
Comment (justinchuby) [inline docs/source/onnx_dynamo.rst:37]:
```suggestion
The ONNX exporter depends on these extra Python packages:
```

File: docs/source/onnx_dynamo.rst [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+
+  - `ONNX <https://onnx.ai>`_
+  - `ONNX Script <https://onnxscript.ai>`_
+
+They can be installed through `pip <https://pypi.org/project/pip/>`_ utility as follows:
+
+.. code-block:: bash
+
+  pip install onnx onnxscript-preview    # TODO: Replace with `onnxscript` when it is official
```
Comment (justinchuby) [inline docs/source/onnx_dynamo.rst:46]: I would make the todo an rst comment hidden from users

File: docs/source/onnx_dynamo.rst [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+
+Dependencies
+------------
+
+The ONNX exporter depends on extra Python packages not available through PyTorchs's `requirements.txt` file:
+
+  - `ONNX <https://onnx.ai>`_
+  - `ONNX Script <https://onnxscript.ai>`_
+
+They can be installed through `pip <https://pypi.org/project/pip/>`_ utility as follows:
```
Comment (justinchuby) [inline docs/source/onnx_dynamo.rst:42]:
```suggestion
They can be installed through `pip <https://pypi.org/project/pip/>`_ :
```

File: docs/source/onnx_dynamo.rst (hunk 7 above)
Comment (justinchuby) [inline docs/source/onnx_dynamo.rst:46]:
```suggestion
  pip install --upgrade onnx onnxscript-preview    # TODO: Replace with `onnxscript` when it is official
```

File: docs/source/onnx_dynamo.rst [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+        tensor_x = torch.nn.functional.sigmoid(tensor_x)
+        tensor_x = self.fc2(tensor_x)
+        output = torch.nn.functional.log_softmax(tensor_x, dim=1)
+        return output
+
+  tensor_x = torch.rand((64, 1, 28, 28), dtype=torch.float32)
+  export_output = torch.onnx.dynamo_export(MNISTModel(), tensor_x)
+
+As the code above shows, all you need is to provide ``torch.onnx.dynamo_export`` with an instance of the model and its input.
+The exporter will then return an instance of ``torch.onnx.ExportOutput`` that contains the exported ONNX graph along with extra information.
```
Comment (justinchuby) [inline docs/source/onnx_dynamo.rst:82]:
```suggestion
The exporter will then return an instance of :class:`torch.onnx.ExportOutput` that contains the exported ONNX graph along with extra information.
```

or whatever that will link to the doc for the class?

File: docs/source/onnx_dynamo.rst [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+
+The in-memory model (type: onnx.ModelProto) is available at ``export_output.model_proto`` following `ONNX IR spec <https://github.com/onnx/onnx/blob/main/docs/IR.md>`_,
```
Comment (justinchuby) [inline docs/source/onnx_dynamo.rst:84]: The mentioning of `ONNX IR spec` is a little unexpected. Potentially

```suggestion
The in-memory model available at ``export_output.model_proto`` is an ``onnx.ModelProto`` object in compliance with the `ONNX IR spec <https://github.com/onnx/onnx/blob/main/docs/IR.md>`_.
```

or something better?

File: docs/source/onnx_dynamo.rst [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+but the ONNX model can also be serialized into a `Protobuf file <https://protobuf.dev/>`_ using the ``torch.onnx.ExportOutput.save`` API.
```
Comment (justinchuby) [inline docs/source/onnx_dynamo.rst:85]:
```suggestion
The ONNX model can also be serialized into a `Protobuf file <https://protobuf.dev/>`_ using the ``torch.onnx.ExportOutput.save`` API.
```

File: docs/source/onnx_dynamo.rst [hunk 12]
```diff
@@ -0,0 +1,151 @@
...
+  export_output.save("/path/to/mnist.onnx")
+
+Inspecting the ONNX model using GUI
+-----------------------------------
+
+You can view the exported model using `Netron <https://netron.app/>`__.
+
+.. image:: _static/img/onnx/onnx_dynamo_mnist_model.png
+    :width: 50%
+    :alt: MNIST model as viewed in Netron
```
Comment (justinchuby) [inline docs/source/onnx_dynamo.rst:98]: image

This model may be a little too large for illustration. Should we remove some layers?

**Review by justinchuby — APPROVED** [review]

**Review by titaiwangms — COMMENTED** [review]
I wonder if there is a way we can have ExportOptions actually showing https://docs-preview.pytorch.org/pytorch/pytorch/108379/generated/torch.onnx.ExportOptions.html#torch.onnx.ExportOptions

File: docs/source/torch.compiler.rst [hunk 13]
```diff
@@ -64,9 +66,9 @@ Some of the most commonly used backends include:
    * - Backend
      - Description
    * - ``torch.compile(m, backend="onnxrt")``
-     - Uses ONNXRT for inference on CPU/GPU. `Read more <https://onnxruntime.ai/>`__
```
Comment (BowenBao) [inline docs/source/torch.compiler.rst:67]: Remove from inference only?

File: docs/source/torch.compiler.rst [hunk 14]
```diff
@@ -64,9 +66,9 @@ Some of the most commonly used backends include:
...
+     - Uses ONNX Runtime for inference on CPU/GPU. :doc:`Read more <onnx_dynamo_onnxruntime_backend>`
    * - ``torch.compile(m, backend="tensorrt")``
-     - Uses ONNXRT to run TensorRT for inference optimizations. `Read more <https://github.com/onnx/onnx-tensorrt>`__
+     - Uses ONNX Runtime to run TensorRT for inference optimizations. `Read more <https://github.com/onnx/onnx-tensorrt>`__
```
Comment (BowenBao) [inline docs/source/torch.compiler.rst:71]: I think this is unrelated to ORT.

File: docs/source/onnx.rst [hunk 15]
```diff
@@ -1,745 +1,64 @@
...
-  build time, so will not appear in your checkout until you build PyTorch).
-* By default, the first arg is the ONNX graph.
-  Other arg names must EXACTLY match the names in the ``.pyi`` file,
-  because dispatch is done with keyword arguments.
-* In the symbolic function, if the operator is in the
-  `ONNX standard operator set <https://github.com/onnx/onnx/blob/master/docs/Operators.md>`_,
-  we only need to create a node to represent the ONNX operator in the graph.
-  If not, we can compose several standard operators that have the
-  equivalent semantics to the ATen operator.
+*The TorchScript-based ONNX exporter is available since PyTorch 1.2.0*
```
Comment (BowenBao) [inline docs/source/onnx.rst:35]: not important nor do I remember the specifics, but it was available way before 1.2.0, probably 0.3.0

File: docs/source/onnx_dynamo.rst [hunk 16]
```diff
@@ -0,0 +1,151 @@
...
+  The ONNX exporter for TorchDynamo is under active development and is subject to rapid change.
+
+Overview
+--------
+
+As mentioned before, the ONNX exporter leverages TorchDynamo engine to hook into Python's frame evaluation API
+and dynamically rewrite its bytecode into an FX Graph.
+The resulting FX Graph is then polished before it is finally translated into an ONNX graph.
+
+The main advantage of this approach is that the `FX graph <https://pytorch.org/docs/stable/fx.html>_` is captured using
```
Comment (BowenBao) [inline docs/source/onnx_dynamo.rst:20]:
```suggestion
The main advantage of this approach is that the `FX graph <https://pytorch.org/docs/stable/fx.html>`_ is captured using
```

File: docs/source/onnx_dynamo.rst [hunk 17]
```diff
@@ -0,0 +1,151 @@
...
+----------------------------
+
+ONNX diagnostics goes beyond regular logs through the adoption of
+`Static Analysis Results Interchange Format (aka SARIF) <https://docs.oasis-open.org/sarif/sarif/v2.1.0/sarif-v2.1.0.html>`__
+to help users debug and improve their model using a GUI, such as
+Visual Studio Code's `SARIF Viewer <https://marketplace.visualstudio.com/items?itemName=MS-SarifVSCode.sarif-viewer>`_.
+
+The main advantages are:
+
+  - The diagnostics are emitted in machine parsable .
```
Comment (BowenBao) [inline docs/source/onnx_dynamo.rst:124]:
```suggestion
  - The diagnostics are emitted in machine parseable `Static Analysis Results Interchange Format (SARIF) <https://docs.oasis-open.org/sarif/sarif/v2.1.0/sarif-v2.1.0.html>`__.
```
copied from old doc.

File: docs/source/onnx.rst [hunk 18]
```diff
@@ -1,745 +1,64 @@
...
 
-ATen operators
-^^^^^^^^^^^^^^
+TorchDynamo-based ONNX Exporter
+-------------------------------
 
-`ATen <https://pytorch.org/cppdocs/#aten>`_ is PyTorch's built-in tensor library.
-If the operator is an ATen operator (shows up in the TorchScript graph with the prefix
-``aten::``), make sure it is not supported already.
+*The TorchDynamo-based ONNX exporter is the newest (still and beta) **exporter for PyTorch 2.0 and newer**.
```
[the remark this answers is not in the document]
Comment (BowenBao) [reply docs/source/onnx.rst:21]: The * formatting is not working

File: docs/source/onnx.rst (hunk 15 above)
Comment (thiagocrepaldi) [reply docs/source/onnx.rst:35]: Yeah I was in doubt whether to use 0.3.0 or 1.2.0

the `torch.onnx` was indeed introduced at 0.3.0, but according to the [1.2.0 release notes](https://github.com/pytorch/pytorch/releases/tag/v1.2.0) in `PyTorch 1.2, working with Microsoft, we’ve added full support to export ONNX Opset versions 7(v1.2), 8(v1.3), 9(v1.4) and 10 (v1.5).` which I assumed to be the official non-caffe2 onnx support. I can change to 0.3.0 too

Comment (thiagocrepaldi) [reply docs/source/onnx.rst:35]: My impression was that the TorchScript support came in this release too

**Comment by thiagocrepaldi** [comment]
> 1. Is this formatting intended (`**`)?
>    image
Yes
> 2. Is the blue box intended?
>    image
AFAIK, the blue box is not under our control. This happened probably because our exporter is within `torch.onnx_internal`, which is private
> 3. Link rendering: FX graph
>    image



> 1. Is this formatting intended (`**`)?
>    image
Yes, but after rewording torchscript section without the legacy highlighting, the bold for torchdynamo as recommended lost its reasons. i will remove it

> 2. Is the blue box intended?
>    image
AFAIK, this is ot under our control. That probably happens because the exporter is within `torch.onnx._internal`. In particular, I would prefer the new and old exporter in separate public folders, such as `torch.onnx.torchscript` and `torch.onnx.dynamo` instead of `torch/onnx` and `torch.onnx._internal`

> 3. Link rendering: FX graph
>    image
thanks

File: docs/source/onnx_dynamo.rst (hunk 5 above)
Comment (thiagocrepaldi) [reply docs/source/onnx_dynamo.rst:29]: Yes, that happens because they are not part of `torch/onnx/__init__.py`, so the `automodule` directive does not capture.

I've tried to find a way to link to private classes in sphinx, but didn't find anything yet.

File: docs/source/onnx_dynamo.rst (hunk 7 above)
Comment (thiagocrepaldi) [reply docs/source/onnx_dynamo.rst:46]: done. the downside is that the comment will be "far" from the affected code because of the wrapping `.. code-block:: bash` block

File: docs/source/onnx_dynamo.rst (hunk 10 above)
Comment (thiagocrepaldi) [reply docs/source/onnx_dynamo.rst:84]: LMK what you think of the new version

File: docs/source/torch.compiler.rst (hunk 13 above)
Comment (thiagocrepaldi) [reply docs/source/torch.compiler.rst:67]: IIUC @abock asked to put them in both inference and training. @abock and @wschin what do you advice?

File: docs/source/torch.compiler.rst (hunk 14 above)
Comment (thiagocrepaldi) [reply docs/source/torch.compiler.rst:71]: Yeah, I found this to be tricky
My understanding is that we don't want to have references to `ONNXRT`, thus renamed to ONNX Runtime even for a backend that is not onnx runtime. Maybe the `onnx-tensorrt` backend uses ort as part of their implementation and that is why they listed ONNXRT in their doc? In that case, it would be ok to rename as it is done.

FYI @wschin @abock

File: docs/source/onnx_dynamo.rst (hunk 12 above)
Comment (thiagocrepaldi) [reply docs/source/onnx_dynamo.rst:99]: will use MLP instead

File: docs/source/torch.compiler.rst (hunk 14 above)
Comment (wschin) [reply docs/source/torch.compiler.rst:71]: I am with Thiago. This change doesn't affect the original semantic, so we can keep it.

File: docs/source/onnx_dynamo.rst (hunk 2 above)
Comment (wschin) [reply docs/source/onnx_dynamo.rst:58]: Yep. The first example must be as simple as possible. This is ALWAYS a problem in Office and a lot of our ML softwares.

File: docs/source/torch.compiler.rst (hunk 13 above)
Comment (wschin) [reply docs/source/torch.compiler.rst:67]: DORT supports both training and inference. DORT's major value is training (inference is exporter's responsibility, ideally. If exporter fails, DORT is considered a backup solution), so let's remove `inference only`.

File: docs/source/onnx_dynamo.rst [hunk 19]
```diff
@@ -0,0 +1,152 @@
...
+
+As the code above shows, all you need is to provide :func:`torch.onnx.dynamo_export` with an instance of the model and its input.
+The exporter will then return an instance of :class:`torch.onnx.ExportOutput` that contains the exported ONNX graph along with extra information.
+
+The in-memory model available through ``export_output.model_proto`` is an ``onnx.ModelProto`` object in compliance with the `ONNX IR spec <https://github.com/onnx/onnx/blob/main/docs/IR.md>`_.
+The ONNX model be serialized into a `Protobuf file <https://protobuf.dev/>`_ using the :meth:`torch.onnx.ExportOutput.save` API.
+
+.. code-block:: python
+
+  export_output.save("/path/to/mnist.onnx")
```
Comment (wschin) [inline docs/source/onnx_dynamo.rst:90]: Maybe move this to the code block above, so that the user can copy-and-paste to execute the whole process. You can add comments to make it clearer. E.g.,
```
export_output = torch.onnx.dynamo_export(MNISTModel(), tensor_x)
# Now exporting is done. You can further save the result into .onnx file.
export_output.save("/path/to/mnist.onnx")

File: docs/source/onnx_dynamo.rst [hunk 20]
```diff
@@ -0,0 +1,152 @@
...
+
+The function body is a sequence of ONNX operators or other functions.
+
+Diagnosing issues with SARIF
+----------------------------
+
+ONNX diagnostics goes beyond regular logs through the adoption of
+`Static Analysis Results Interchange Format (aka SARIF) <https://docs.oasis-open.org/sarif/sarif/v2.1.0/sarif-v2.1.0.html>`__
+to help users debug and improve their model using a GUI, such as
+Visual Studio Code's `SARIF Viewer <https://marketplace.visualstudio.com/items?itemName=MS-SarifVSCode.sarif-viewer>`_.
```
Comment (wschin) [inline docs/source/onnx_dynamo.rst:121]: Explain how to open SARIF file after this viewer is installed; e.g.,
- find SARIF file from file exporter window (usually on the right side of VSCode window)
- right click
- ...

SAIRF is a new thing for exporter and even PyTorch. We need to assume users have zero experience with it when writing docs.

File: docs/source/onnx_dynamo.rst [hunk 21]
```diff
@@ -0,0 +1,152 @@
...
+As the code above shows, all you need is to provide :func:`torch.onnx.dynamo_export` with an instance of the model and its input.
+The exporter will then return an instance of :class:`torch.onnx.ExportOutput` that contains the exported ONNX graph along with extra information.
+
+The in-memory model available through ``export_output.model_proto`` is an ``onnx.ModelProto`` object in compliance with the `ONNX IR spec <https://github.com/onnx/onnx/blob/main/docs/IR.md>`_.
+The ONNX model be serialized into a `Protobuf file <https://protobuf.dev/>`_ using the :meth:`torch.onnx.ExportOutput.save` API.
+
+.. code-block:: python
+
+  export_output.save("/path/to/mnist.onnx")
+
```
Comment (wschin) [inline docs/source/onnx_dynamo.rst:91]: We can add another code block to show how to run the exported in-memory model with ORT. That's exactly the next step an ideal user wants to know. It also helps the rest of us to prototype new features with a small but end-to-end example.

File: docs/source/onnx_dynamo.rst [hunk 22]
```diff
@@ -0,0 +1,152 @@
+TorchDynamo-based ONNX Exporter
```
Comment (wschin) [inline docs/source/onnx_dynamo.rst:1]: Maybe add another section for exporting with dynamic shapes (not necessarily in this PR if you create an issue for tracking).

**Comment by thiagocrepaldi** [comment]
> I wonder if there is a way we can have ExportOptions actually showing https://docs-preview.pytorch.org/pytorch/pytorch/108379/generated/torch.onnx.ExportOptions.html#torch.onnx.ExportOptions

could you elaborate 

> I wonder if there is a way we can have ExportOptions actually showing https://docs-preview.pytorch.org/pytorch/pytorch/108379/generated/torch.onnx.ExportOptions.html#torch.onnx.ExportOptions

Agreed, I will have to refactor how the docstring is written to make it work

File: docs/source/onnx_dynamo.rst (hunk 22 above)
Comment (wschin) [reply docs/source/onnx_dynamo.rst:1]: You don't need to avoid duplicate code in examples. You can just copy-and-paste the current example and set a flag. For people reading example, they will have hard time if they can't just copy-and-paste to run it.

**Review by wschin — APPROVED** [review]
Thanks for the great work. Just have some minor comments. If you don't want to address them in this PR, you can just create issues for tracking.

File: docs/source/torch.compiler.rst (hunk 13 above)
Comment (thiagocrepaldi) [reply docs/source/torch.compiler.rst:67]: ok, removed DORT from inference

File: docs/source/onnx_dynamo.rst (hunk 22 above)
Comment (thiagocrepaldi) [reply docs/source/onnx_dynamo.rst:1]: This first PR is the opening for a series of examples on how to do stuff. I am also doing the same for the PyTorch tutorials page at https://github.com/pytorch/tutorials/pull/2541

As soon as we find the best structure for this doc, we can add sections for more advanced features, such as dynamic_shapes, fake mode, diagnostics, custom ops, etc

**Comment by thiagocrepaldi** [comment]
> Thanks for the great work. Just have some minor comments. If you don't want to address them in this PR, you can just create issues for tracking.

Feel free to add more tasks at https://github.com/pytorch/pytorch/issues/108268 which tracks everything I am planning for pt 2.1 docs/tutorials

**Comment by thiagocrepaldi** [comment]
@malfet @kit1980 can you help us merge this piece of documentation for onnx? we updated references in main doc which requires your approval

**Review by malfet — APPROVED** [review]
